### PR TITLE
add support for int key in bucket level trigger aggregations

### LINKED ISSUE #311: How to get access the key from aggregated bucket with bucket_level_trigger？
I was try to create an alert as below

```
POST _plugins/_alerting/monitors
{
  "type": "monitor",
  "name": "alert-demo-6",
  "monitor_type": "bucket_level_monitor",
  "enabled": true,
  "schedule": {
    "period": {
      "interval": 1,
      "unit": "MINUTES"
    }
  },
  "inputs": [
    {
      "search": {
        "indices": [
          "sample-http-responses"
        ],
        "query": {
          "size": 1,
          "query": {
            "match_all": {}
          },
          "aggs": {
            "status_code": {
              "terms": {
                "field": "status_code",
                "size": 10
              }
            }
          }
        }
      }
    }
  ],
  "triggers": [
    {
      "bucket_level_trigger": {
        "name": "trigger-demo-6",
        "severity": "1",
        "condition": {
          "buckets_path": {
            "_count":"_count",
            "_key": "_key"
          },
          "parent_bucket_path": "status_code",
          "script": {
            "source": """
                params._count > 0

            """,
            "lang": "painless"
          }
        },
        "actions": [
          {
            "name": "test-action",
            "destination_id": "lCsf-34BSFe9yDgK4FWH",
            "message_template": {
              "source": """{
                  "Monitor": "{{ctx.monitor.name}}",
                  "Severity": "{{ctx.trigger.severity}}",
                  "Period start": "{{ctx.periodStart}}",
                  "Period end": "{{ctx.periodEnd}}",
                  "Error url": "{{ctx.results.0.aggregations.status_code.buckets.0.key}}",
                  "Error count": "{{ctx.results.0.aggregations.status_code.buckets.0.doc_count}}",
                  "team": "aaa",
                  "support owner": "izebbbbbk"
                  }
              """,
              "lang": "mustache"
            },
            "throttle_enabled": false,
            "throttle": {
              "value": 10,
              "unit": "MINUTES"
            },
            "action_execution_policy": {
              "action_execution_scope": {
                "per_alert": {
                  "actionable_alerts": [
                    "DEDUPED",
                    "NEW"
                  ]
                }
              }
            },
            "subject_template": {
              "source": "alert-demo-4",
              "lang": "mustache"
            }
          }
        ]
      }
    }
  ]
}
```

In the bucket_level_trigger section, I set the buckets_path to  "_count":"_count","_key": "_key", then after execute the alert, it shows the error "Unexpected format for key in bucket [{doc_count=4583, key=200}]"

```

{
  "monitor_name" : "alert-demo-6",
  "period_start" : 1645169113039,
  "period_end" : 1645169173039,
  "error" : null,
  "input_results" : {
    "results" : [
      {
        "_shards" : {
          "total" : 1,
          "failed" : 0,
          "successful" : 1,
          "skipped" : 0
        },
        "hits" : {
          "hits" : [
            {
              "_index" : "sample-http-responses",
              "_type" : "_doc",
              "_source" : {
                "status_code" : 100,
                "http_4xx" : 0,
                "http_3xx" : 0,
                "http_5xx" : 0,
                "http_2xx" : 0,
                "timestamp" : 1643772700360,
                "http_1xx" : 1
              },
              "_id" : "0",
              "_score" : 1.0
            }
          ],
          "total" : {
            "value" : 10000,
            "relation" : "gte"
          },
          "max_score" : 1.0
        },
        "took" : 4,
        "timed_out" : false,
        "aggregations" : {
          "status_code" : {
            "doc_count_error_upper_bound" : 0,
            "sum_other_doc_count" : 53,
            "buckets" : [
              {
                "doc_count" : 4583,
                "key" : 200
              },
              {
                "doc_count" : 4501,
                "key" : 301
              },
              {
                "doc_count" : 4498,
                "key" : 202
              },
              {
                "doc_count" : 4486,
                "key" : 101
              },
              {
                "doc_count" : 4484,
                "key" : 201
              },
              {
                "doc_count" : 4462,
                "key" : 302
              },
              {
                "doc_count" : 4457,
                "key" : 300
              },
              {
                "doc_count" : 4427,
                "key" : 100
              },
              {
                "doc_count" : 4354,
                "key" : 102
              },
              {
                "doc_count" : 15,
                "key" : 404
              }
            ]
          },
          "k9a5C38BMQQOgIcxJnPA" : {
            "parent_bucket_path" : "status_code",
            "bucket_indices" : [
              0,
              1,
              2,
              3,
              4,
              5,
              6,
              7,
              8,
              9
            ]
          }
        }
      }
    ],
    "error" : null
  },
  "trigger_results" : {
    "k9a5C38BMQQOgIcxJnPA" : {
      "name" : "trigger-demo-6",
      "agg_result_buckets" : { },
      "action_results" : { },
      "error" : "Unexpected format for key in bucket [{doc_count=4583, key=200}]"
    }
  }
}

```

How can I access each Key from the bucket_indices?

The goal I would like to achieve here is when 404 higher then certain number.
Therefore, I would like to have script within the condition part as below

if(params._key == 404) {
    if(params._count >10) {
        return true;
   }
}
return false;

How can I achieve this, please?

### PR DESCRIPTION
**Pull request by sbcd90**
Signed-off-by: Subhobrata Dey <sbcd90@gmail.com>

*Issue #, if available:*
fixes #311 

*Description of changes:*
Add support for int datatype as bucket keyField

*CheckList:*
[X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: alerting/src/test/kotlin/org/opensearch/alerting/TriggerServiceTests.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        )
+        val trigger = randomBucketLevelTrigger(bucketSelector = bucketSelectorExtAggregationBuilder)
+        val monitor = randomBucketLevelMonitor(triggers = listOf(trigger))
+
+        val inputResultsStr = "{\"_shards\":{\"total\":1,\"failed\":0,\"successful\":1,\"skipped\":0},\"hits\":{\"hits\":[{\"_index\":\"sample-http-responses\",\"_type\":\"http\",\"_source\":{\"status_code\":100,\"http_4xx\":0,\"http_3xx\":0,\"http_5xx\":0,\"http_2xx\":0,\"timestamp\":100000,\"http_1xx\":1},\"_id\":1,\"_score\":1}],\"total\":{\"value\":4,\"relation\":\"eq\"},\"max_score\":1},\"took\":37,\"timed_out\":false,\"aggregations\":{\"status_code\":{\"doc_count_error_upper_bound\":0,\"sum_other_doc_count\":0,\"buckets\":[{\"doc_count\":2,\"key\":100},{\"doc_count\":1,\"key\":102},{\"doc_count\":1,\"key\":201}]},\"${trigger.id}\":{\"parent_bucket_path\":\"status_code\",\"bucket_indices\":[0,1,2]}}}"
+
+        val xContentType = XContentType.fromMediaTypeOrFormat(ContentType.APPLICATION_JSON.toString())
+        val parser = xContentType.xContent().createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION, inputResultsStr)
```
Comment (qreshi) [inline alerting/src/test/kotlin/org/opensearch/alerting/TriggerServiceTests.kt:44]: I think you can just do `XContentType.JSON.xContent().createParser` here

File: alerting/src/test/kotlin/org/opensearch/alerting/TriggerServiceTests.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    private lateinit var scriptService: ScriptService
+    private lateinit var triggerService: TriggerService
+
+    @Before
+    fun setup() {
+        scriptService = Mockito.mock(ScriptService::class.java)
+        triggerService = TriggerService(scriptService)
+    }
+
+    fun `test Run Bucket Level Triggers`() {
```
Comment (qreshi) [inline alerting/src/test/kotlin/org/opensearch/alerting/TriggerServiceTests.kt:32]: Can we name this test to `test run bucket level trigger with bucket key as int` to be more specific.

Also, did you happen to check something like a composite aggregation containing an aggregation on int? In that case we'd hit the `bucket[keyField] is Map<*, *>` case and convert it into a String which I'm assuming may behave strangely and not be fully supported?

**Comment by sbcd90** [comment]
Hi @qreshi @rishabhmaurya , fixed the code review comments. Added a test case for composite aggregation case.
Found a new issue while testing `composite aggregation`.

Stacktrace:

```
org.opensearch.action.search.SearchPhaseExecutionException: 
	at org.opensearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:642) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.action.search.FetchSearchPhase$1.onFailure(FetchSearchPhase.java:126) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:57) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:792) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[?:?]
	at java.lang.Thread.run(Thread.java:832) ~[?:?]
Caused by: java.lang.IllegalStateException: value must not be null
	at org.opensearch.alerting.aggregation.bucketselectorext.BucketSelectorExtAggregator.doReduce(BucketSelectorExtAggregator.kt:135) ~[?:?]
	at org.opensearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:230) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:561) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:532) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:151) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:135) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.action.search.FetchSearchPhase.access$000(FetchSearchPhase.java:60) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:121) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
	... 6 more
```

Initial debugging suggests its an `opensearch-core` issue.

**Comment by qreshi** [comment]
> Hi @qreshi @rishabhmaurya , fixed the code review comments. Added a test case for composite aggregation case. Found a new issue while testing `composite aggregation`.
> 
> Stacktrace:
> 
> ```
> org.opensearch.action.search.SearchPhaseExecutionException: 
> 	at org.opensearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:642) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.action.search.FetchSearchPhase$1.onFailure(FetchSearchPhase.java:126) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:57) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:792) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[?:?]
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[?:?]
> 	at java.lang.Thread.run(Thread.java:832) ~[?:?]
> Caused by: java.lang.IllegalStateException: value must not be null
> 	at org.opensearch.alerting.aggregation.bucketselectorext.BucketSelectorExtAggregator.doReduce(BucketSelectorExtAggregator.kt:135) ~[?:?]
> 	at org.opensearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:230) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:561) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:532) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:151) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:135) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.action.search.FetchSearchPhase.access$000(FetchSearchPhase.java:60) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:121) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50) ~[opensearch-1.2.4-SNAPSHOT.jar:1.2.4-SNAPSHOT]
> 	... 6 more
> ```
> 
> Initial debugging suggests its an `opensearch-core` issue.

Looks like you're using `opensearch-1.2.4-SNAPSHOT` there. Any reason it's not on OS 2.0 or at least OS 1.3 snapshot?

**Comment by sbcd90** [comment]
@qreshi sure I'll test with OS 1.3 snapshot & see if the same issue still exists.

**Comment by sbcd90** [comment]
hi @qreshi , i produced the following issue with OS 1.3 & looks like the code that is causing this issue is still in the os-core `main` branch.

create index `sample-http-responses`. insert data from here: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/server/sampleData/rawData/httpResponses.json.gz

In `Opensearch-dashboards` -> `Create monitor` -> `per bucket monitor` -> `extraction query editor` -> use query

```
{
          "size": 1,
          "query": {
            "match_all": {}
          },
          "aggs": {
            "status_code": {
                "composite": {
                    "sources": [
                        { "status_code": { "terms": { "field": "status_code" } } }
                    ]
                }
            }
          }
        }
``` 
& then `add trigger` -> `trigger condition` add this

```
{
          "buckets_path": {
            "_count":"_count",
            "_key": "_key"
          },
          "parent_bucket_path": "status_code",
          "script": {
            "source": "params._count > 0",
            "lang": "painless"
          }
        }
```
Now click on `preview condition response` gives error:
```
Empty list doesn't contain element at index 0.
```

Actual error in the logs:
```
org.opensearch.action.search.SearchPhaseExecutionException: 
	at org.opensearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:642) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.action.search.FetchSearchPhase$1.onFailure(FetchSearchPhase.java:126) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:52) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:57) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:792) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
	at java.lang.Thread.run(Thread.java:829) ~[?:?]
Caused by: org.opensearch.search.aggregations.AggregationExecutionException: buckets_path must reference either a number value or a single value numeric metric aggregation, got: [ArrayMap] at aggregation [_key]
	at org.opensearch.search.aggregations.pipeline.BucketHelpers.formatResolutionError(BucketHelpers.java:273) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.search.aggregations.pipeline.BucketHelpers.resolveBucketValue(BucketHelpers.java:202) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.search.aggregations.pipeline.BucketHelpers.resolveBucketValue(BucketHelpers.java:178) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.alerting.aggregation.bucketselectorext.BucketSelectorExtAggregator.doReduce(BucketSelectorExtAggregator.kt:132) ~[?:?]
	at org.opensearch.search.aggregations.InternalAggregations.topLevelReduce(InternalAggregations.java:230) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.action.search.SearchPhaseController.reduceAggs(SearchPhaseController.java:561) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.action.search.SearchPhaseController.reducedQueryPhase(SearchPhaseController.java:532) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.action.search.QueryPhaseResultConsumer.reduce(QueryPhaseResultConsumer.java:151) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:135) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.action.search.FetchSearchPhase.access$000(FetchSearchPhase.java:60) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:121) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	at org.opensearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:50) ~[opensearch-1.3.0-SNAPSHOT.jar:1.3.0-SNAPSHOT]
	... 6 more
[2022-03-17T21:47:51,079][INFO ][o.o.a.TriggerService     ] [dev-dsk-deysubho-2a-cc6dd507.us-west-2.amazon.com] Error running trigger [KsLZmX8BYDwqmLCh73id] for monitor []
java.lang.IndexOutOfBoundsException: Empty list doesn't contain element at index 0.
	at kotlin.collections.EmptyList.get(Collections.kt:35) ~[kotlin-stdlib-1.3.72.jar:1.3.72-release-468 (1.3.72)]
	at kotlin.collections.EmptyList.get(Collections.kt:23) ~[kotlin-stdlib-1.3.72.jar:1.3.72-release-468 (1.3.72)]
	at org.opensearch.alerting.TriggerService.runBucketLevelTrigger(TriggerService.kt:64) [opensearch-alerting-1.3.0.0-SNAPSHOT.jar:1.3.0.0-SNAPSHOT]
	at org.opensearch.alerting.MonitorRunner.runBucketLevelMonitor(MonitorRunner.kt:384) [opensearch-alerting-1.3.0.0-SNAPSHOT.jar:1.3.0.0-SNAPSHOT]
	at org.opensearch.alerting.MonitorRunner$runBucketLevelMonitor$1.invokeSuspend(MonitorRunner.kt) [opensearch-alerting-1.3.0.0-SNAPSHOT.jar:1.3.0.0-SNAPSHOT]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33) [kotlin-stdlib-1.3.72.jar:1.3.72-release-468 (1.3.72)]
	at kotlinx.coroutines.ResumeModeKt.resumeUninterceptedMode(ResumeMode.kt:46) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlinx.coroutines.internal.ScopeCoroutine.onCompletionInternal$kotlinx_coroutines_core(Scopes.kt:28) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlinx.coroutines.JobSupport.completeStateFinalization(JobSupport.kt:305) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlinx.coroutines.JobSupport.tryFinalizeSimpleState(JobSupport.kt:264) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlinx.coroutines.JobSupport.tryMakeCompleting(JobSupport.kt:762) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlinx.coroutines.JobSupport.makeCompletingOnce$kotlinx_coroutines_core(JobSupport.kt:742) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlinx.coroutines.AbstractCoroutine.resumeWith(AbstractCoroutine.kt:117) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:46) [kotlin-stdlib-1.3.72.jar:1.3.72-release-468 (1.3.72)]
	at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:285) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60) [kotlinx-coroutines-core-1.1.1.jar:?]
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:742) [kotlinx-coroutines-core-1.1.1.jar:?]
```

**Comment by qreshi** [comment]
@sbcd90 Thanks for the additional info. Also, what was the response of the extraction query itself?

**Comment by sbcd90** [comment]
hi @qreshi , here is the response
```
{
    "_shards": {
        "total": 1,
        "failed": 0,
        "successful": 1,
        "skipped": 0
    },
    "hits": {
        "hits": [
            {
                "_index": "sample-http-responses",
                "_type": "http",
                "_source": {
                    "status_code": 100,
                    "http_4xx": 0,
                    "http_3xx": 0,
                    "http_5xx": 0,
                    "http_2xx": 0,
                    "timestamp": 100000,
                    "http_1xx": 1
                },
                "_id": "1",
                "_score": 1
            }
        ],
        "total": {
            "value": 4,
            "relation": "eq"
        },
        "max_score": 1
    },
    "took": 3,
    "timed_out": false,
    "aggregations": {
        "status_code": {
            "buckets": [
                {
                    "doc_count": 2,
                    "key": {
                        "status_code": 100
                    }
                },
                {
                    "doc_count": 1,
                    "key": {
                        "status_code": 102
                    }
                },
                {
                    "doc_count": 1,
                    "key": {
                        "status_code": 201
                    }
                }
            ],
            "after_key": {
                "status_code": 201
            }
        }
    }
}
```

**Comment by sbcd90** [comment]
As per discussions with @qreshi & @rishabhmaurya & after debugging `opensearch-core`, looks like `group by key` is not supported as a feature in `opensearch-core` for composite aggregations. 

However, there is a way to make `group by key` work in alerting using composite aggregations filters.

```
{
          "buckets_path": {
            "_count":"_count"
          },
          "parent_bucket_path": "status_code",
          "script": {
            "source": "params._count > 0",
            "lang": "painless"
          },
          "composite_agg_filter":{"status_code":{"include":[],"exclude":[]}}
}
```

**Review by qreshi — APPROVED** [review]

**Review by rishabhmaurya — APPROVED** [review]

**Pull request merged**